python/pylibcudf/pylibcudf/scalar.pyx

```diff
@@ -744,16 +744,14 @@ if np is not None:
         return slr
 
 
-if pa is not None:
-    def _from_arrow(obj: pa.Scalar, dtype: DataType | None = None) -> Scalar:
-        if isinstance(obj.type, pa.ListType) and obj.as_py() is None:
-            # pyarrow doesn't correctly handle None values for list types, so
-            # we have to create this one manually.
-            # https://github.com/apache/arrow/issues/40319
-            pa_array = pa.array([None], type=obj.type)
-        else:
-            pa_array = pa.array([obj])
-        return Column.from_arrow(pa_array, dtype=dtype).to_scalar()
-else:
-    def _from_arrow(obj: pa.Scalar, dtype: DataType | None = None) -> Scalar:
+def _from_arrow(obj: pa.Scalar, dtype: DataType | None = None) -> Scalar:
```

Comment:
I'm surprised that mypy or something else doesn't complain about you using `pa.Scalar` to type this when `pa` could be None. Do you know how this is working?

Reply:
I don't, unfortunately - my knowledge of mypy interacting with cython files generally is limited. 

I generally considered the annotations here to be "documenting annotations".